proxy/src/main/java/com/velocitypowered/proxy/bossbar/VelocityBossBar.java

```diff
@@ -0,0 +1,216 @@
+  }
+
+  @Override
+  public @NonNull Component getTitle() {
+    return title;
+  }
+
+  @Override
+  public void setTitle(@NonNull Component title) {
+    this.title = title;
+    BossBar bar = new BossBar();
+    bar.setUuid(uuid);
+    bar.setAction(BossBar.UPDATE_NAME);
+    bar.setName(GsonComponentSerializer.INSTANCE.serialize(title));
+    players.forEach(
+        player -> {
+          if (player.isActive() && visible) {
+            sendPacket(player, bar);
+          }
+        });
+  }
+
+  @Override
+  public float getProgress() {
+    return progress;
+  }
+
+  @Override
+  public void setProgress(float progress) {
+    if (progress > 1 || progress < 0) {
+      throw new IllegalArgumentException("Progress not between 0 and 1");
+    }
+    this.progress = progress;
+    BossBar bar = new BossBar();
+    bar.setUuid(uuid);
+    bar.setAction(BossBar.UPDATE_PERCENT);
+    bar.setPercent(progress);
+    players.forEach(
+        player -> {
+          if (player.isActive() && visible) {
+            sendPacket(player, bar);
+          }
+        });
+  }
+
+  @Override
+  public @Nullable Collection<Player> getPlayers() {
+    return Collections.unmodifiableCollection(players);
+  }
+
+  @Override
+  public @NonNull UUID getUUID() {
+    return uuid;
+  }
+
+  @Override
+  public @NonNull BossBarColor getColor() {
+    return color;
+  }
+
+  @Override
+  public void setColor(@NonNull BossBarColor color) {
+    this.color = color;
+    setDivisions(color, overlay);
+  }
+
+  @Override
+  public @NonNull BossBarOverlay getOverlay() {
+    return overlay;
+  }
+
+  @Override
+  public void setOverlay(@NonNull BossBarOverlay overlay) {
+    this.overlay = overlay;
+    setDivisions(color, overlay);
+  }
+
+  private void setDivisions(BossBarColor color, BossBarOverlay overlay) {
+    BossBar bar = new BossBar();
+    bar.setUuid(uuid);
+    bar.setAction(BossBar.UPDATE_STYLE);
+    bar.setColor(color.ordinal());
+    bar.setOverlay(overlay.ordinal());
+    players.forEach(
+        player -> {
+          if (player.isActive() && visible) {
+            sendPacket(player, bar);
+          }
+        });
+  }
+
+  @Override
+  public boolean isVisible() {
+    return visible;
+  }
+
+  @Override
+  public void setVisible(boolean visible) {
+    if (!visible) {
+      removeAllPlayers();
+    }
+    this.visible = visible;
+  }
+
+  private BossBar addPacket() {
+    BossBar bossBar = new BossBar();
+    bossBar.setUuid(uuid);
+    bossBar.setAction(BossBar.ADD);
+    bossBar.setName(GsonComponentSerializer.INSTANCE.serialize(title));
+    bossBar.setColor(color.ordinal());
+    bossBar.setOverlay(overlay.ordinal());
+    bossBar.setPercent(progress);
+    return bossBar;
+  }
+
+  private BossBar removePacket() {
+    BossBar bossBar = new BossBar();
+    bossBar.setUuid(uuid);
+    bossBar.setAction(BossBar.REMOVE);
+    return bossBar;
+  }
+
+  private void sendPacket(Player player, MinecraftPacket packet) {
```

Comment:
This method shouldn't go in this class

Reply:
I just do it to send the packets easier. If you ask me, Velocity should add the protocol things to the api and add method for sending packets in the player object

Reply:
Adding "protocol things" to the API won't happen. It is a sign that we need to extend the API instead.

Reply:
Its not to the api. -_-